### PR TITLE
Add Installation UUID to Usage Data reports

### LINKED ISSUE #493: Add installation uuid
We get usage data, but can't clearly correlate two sessions other than by heuristics.  Ideally we can uniquely identify when things are happening within a single install.

E.g., I can only count platforms by board (which is accurate if they only ever use one, and absent if they don't get a flight controller working).

### PR DESCRIPTION
**Pull request by jpbarraca**
Creates an installation UUID in the configuration file and reports this UUID together with usage data. The UUID should be constant over different builds as long as settings are not erased.

[Review on Reviewable](https://reviewable.io/reviews/d-ronin/dronin/612)

edit by MPL: Fixes #493

### CODE REVIEW

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by dustin** [comment]
Thanks, this looks like exactly what I wanted.

:lgtm:

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/612#-:-K9hZvJhcr3ssjsWEIag:-137267799)_

**Comment by mlyle** [comment]
@jpbarraca--- thanks for the PR, it looks great.  We'll give it a try and get it merged in the next couple of days.  It'll be a big help in understanding how people use the software.

**Comment by mlyle** [comment]
I have not had a chance to look at this yet in detail, but it looks like this is what shows up in my usage report right now:

```
"installationUUID":""
```

Also it would be better if the SDL change was in a separate PR-- do you need any help knowing how to rebase that out of this PR and get it in another one?

**Comment by jpbarraca** [comment]
My bad. The branch is ok now. Will check the empty UUID.

**Comment by jpbarraca** [comment]
Fixed. Please check.

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
Works great, thank you!   Will merge once it gets through the Jenkins build.

**Comment by mlyle** [comment]
jenkins, test this please (underlying problem fixed)

**Comment by PTDreamer** [comment]
I know this was already merged but I will still add some comments about some issues that should be fixed in the future.

File: ground/gcs/src/app/main.cpp [hunk 1]
```diff
@@ -292,6 +293,7 @@ int main(int argc, char **argv)
         }
         overrideSettings(parser, settings);
         locale = settings->value("General/OverrideLanguage", locale).toString();
+
```
Comment (PTDreamer) [inline ground/gcs/src/app/main.cpp:296]: New line on unrelated code should be removed

File: ground/gcs/src/app/main.cpp [hunk 2]
```diff
@@ -60,6 +60,7 @@
 #include <QBitmap>
 #include <QCommandLineParser>
 #include <QCommandLineOption>
+#include <QUuid>
```
Comment (PTDreamer) [inline ground/gcs/src/app/main.cpp:63]: This should not be included in main.

File: ground/gcs/src/plugins/usagestatsgadget/usagestatsplugin.cpp [hunk 3]
```diff
@@ -64,6 +64,14 @@ void UsageStatsPlugin::readConfig(QSettings *qSettings, Core::UAVConfigInfo *con
     qSettings->beginGroup(QLatin1String("UsageStatistics"));
     sendUsageStats = (qSettings->value(QLatin1String("SendUsageStats"), sendUsageStats).toBool());
     sendPrivateData = (qSettings->value(QLatin1String("SendPrivateData"), sendPrivateData).toBool());
+
+    //Check the Installation UUID and Generate a new one if required
+    installationUUID = QUuid(qSettings->value(QLatin1String("InstallationUUID"),"").toString());
```
Comment (PTDreamer) [inline ground/gcs/src/plugins/usagestatsgadget/usagestatsplugin.cpp:69]: Missing spaces throughout not complying with current coding stile.

File: ground/gcs/src/plugins/usagestatsgadget/usagestatsplugin.cpp [hunk 4]
```diff
@@ -235,6 +243,10 @@ QByteArray UsageStatsPlugin::processJson() {
     json["currentOS"] = QSysInfo::prettyProductName();
     json["currentArch"] = QSysInfo::currentCpuArchitecture();
     json["buildInfo"] = QSysInfo::buildAbi();
+
+    if(!installationUUID.isNull())
+        json["installationUUID"] = installationUUID.toString().remove(QRegExp("[{}]*"));
```
Comment (PTDreamer) [inline ground/gcs/src/plugins/usagestatsgadget/usagestatsplugin.cpp:248]: Should use the new getInstallationUUID function to avoid code duplication.

**Comment by PTDreamer** [comment]
Since this brings a new dependency all packages should be tested.

**Comment by jpbarraca** [comment]
Thanks for the review. ~~What is the current coding style?~~
